src/opencl/utils.rs

```diff
@@ -54,6 +71,14 @@ fn build_device_list() -> Vec<Device> {
     let mut all_devices = Vec::new();
     let platforms: Vec<_> = opencl3::platform::get_platforms().unwrap_or_default();
 
+    // Laptops might have an integrated GPU. Such devices might have neither a PCI-ID, nor a UUID.
+    // As those devices are used for development and not for production use, it's good enough to
+    // provide a workaround which doesn't add much complexity to the code. We use a fake PCI-ID
+    // instead, which is generated by enumerating the available devices. In order to make that
+    // case easier to spot when debugging issues, a starting number which is pleasant to the human
+    // eye was choosen, that works both, decimal and hexadecimal (4660 == 0x1234).
+    let mut enumerated_device: u16 = 4660;
```

Comment:
what if both are the case, the code should probably check at least for conflicts with any existing PCI-ID